build_matrix_A.py

```diff
@@ -0,0 +1,91 @@
+import numpy as np
+
+
+def build_matrix_A(material_property_library, mesh, time):
+    """ This function will take a previously generated mesh and create the 'A'
+    matrix used in solving this problem.
+
+    inputs:
+    --------
+    - material_property_library: this will be a dictionary or nested dictionary
+    containing all of the important data for each material (fuel,
+    various layers of graphite)
+    - mesh: the physical location of each mesh point
+    - materials: the materials in each region and how many nodes are assigned to that material
+    outputs. List of tuples of format (material, number of nodes)
+    - time: An array of timesteps taken during transients. If steady state, supply 0.
+    --------
+    - An array that can be plugged into a subsequent function to solve the
+      A.x=b formula
+    """
+    k = np.zeros(len(mesh))
+    rho = np.zeros(len(mesh))
+    c = np.zeros(len(mesh))
+    DR = np.zeros(len(mesh))
+    if len(time) > 1:
```

Comment:
Why not just pass in `Dt` since that's the only purpose for `time`?  It could have a default value of 1 and then only be passed in if necessary.

Reply:
I passed in the full time array because Dt is not calculated explicitly except for the matrix generators, so it seemed equivalent to me.